commands/plugin_bot.go

```diff
@@ -186,6 +186,23 @@ func YAGCommandMiddleware(inner dcmd.RunFunc) dcmd.RunFunc {
 			}
 		}
 
+		// Check if the user can execute the command
+		canExecute, resp, settings, err := yc.checkCanExecuteCommand(data)
+		if err != nil {
+			yc.Logger(data).WithError(err).Error("An error occured while checking if we could run command")
+		}
+
+		parseErr := dcmd.ParseCmdArgs(data)
```

Comment:
It's idiomatic to use the name `err` for errors throughout, so perhaps reassign `err = dcmd.ParseCmdArgs(data)` instead if possible?